src/structs/truncate_components.rs

```diff
@@ -10,14 +10,14 @@ use std::cmp;
 
```

Comment:
In principle, all this structure is also a duplication of `ComponentInfo` and we can think about merging them.

Reply:
Yeah, next change :)